astroquery/eso/core.py

```diff
@@ -296,248 +218,472 @@ def _login(self, *, username: str = None, store_password: bool = False,
-            other_collections = root.find('select', id='collection_name_option')
-            # it is possible to have empty collections or other collections...
-            collection_elts = (collections_table.find_all('input', type='checkbox')
-                               if collections_table is not None
-                               else [])
-            other_elts = (other_collections.find_all('option')
-                          if other_collections is not None
-                          else [])
-            for element in (collection_elts + other_elts):
-                if 'value' in element.attrs:
-                    survey = element.attrs['value']
-                    if survey and survey not in self._survey_list and 'Any' not in survey:
-                        self._survey_list.append(survey)
-        return self._survey_list
-
-    def query_surveys(self, *, surveys='', cache=True,
-                      help=False, open_form=False, **kwargs):
+        _ = cache  # We're aware about disregarding the argument
+        t = _EsoNames.phase3_table
+        c = _EsoNames.phase3_surveys_column
+        query_str = f"select distinct {c} from {t}"
+        res = list(self.query_tap_service(query_str)[c].data)
+        return res
+
+    @unlimited_max_rec
+    def _print_table_help(self, table_name: str) -> None:
+        """
+        Prints the columns contained in a given table
+        """
+        help_query = (
+            f"select column_name, datatype, xtype, unit "
+            # TODO: The column description renders output unmanageable
+            # f", description "
+            f"from TAP_SCHEMA.columns "
+            f"where table_name = '{table_name}'")
+        available_cols = self.query_tap_service(help_query)
+
+        count_query = f"select count(*) from {table_name}"
+        num_records = list(self.query_tap_service(count_query)[0].values())[0]
+
+        # All this block is to print nicely...
+        # This whole function should be better written and the output
+        # shown tidier
+        nlines = len(available_cols) + 2
+        n_ = astropy.conf.max_lines
+        m_ = astropy.conf.max_width
+        astropy.conf.max_lines = nlines
+        astropy.conf.max_width = sys.maxsize
+        log.info(f"\nColumns present in the table {table_name}:\n{available_cols}\n"
+                 f"\nNumber of records present in the table {table_name}:\n{num_records}\n")
+        astropy.conf.max_lines = n_
+        astropy.conf.max_width = m_
+
+    def _query_on_allowed_values(
+        self,
+        user_params: _UserParams
+    ) -> Union[astropy.table.Table, int, str]:
+        up = user_params  # shorthand
+
+        if up.print_help:
+            self._print_table_help(up.table_name)
+            return
+
+        _raise_if_has_deprecated_keys(up.column_filters)
+
+        raise_if_coords_not_valid(up.cone_ra, up.cone_dec, up.cone_radius)
+
+        query = _py2adql(up)
+
+        if up.query_str_only:
```

Comment:
have a `get_query_payload` argument instead of this to be consistent with the rest of the API